samples/SimpleDataSource/HelloWorldCustomDataProcessor.cs

```diff
@@ -0,0 +1,159 @@
+        {
+            //
+            // This is where you add your own logic to process the data into a format for your tables.
+            //
+            // In this sample, our tables will operate on a collection of lines in the file
+            // so we read all of the lines of the file and store them into a backing dictionary field.
+            //
+            // ProcessAsync is also where you would determine the information necessary for GetDataSourceInfo().
+            // In this sample, we take down the start and stop timestamps from the files
+            //
+            // Note: if you must do processing based on which tables are enabled, you would check the EnabledTables property (provided in the base class) on your class to see what you should do.
+            // For example, XPerf looks at what tables are enabled in order to turn on additional event processors to avoid processing everything if it doesn't have to.
+            //
+            // In this sample we create a Time Stamp column that can be used for graphing. For more information on graphing, go to "Graphing" on our Wiki. Link can be found in README.md
+            //
+
+            Timestamp startTime = Timestamp.MaxValue;
+            Timestamp endTime = Timestamp.MinValue;
+            DateTime firstEvent = DateTime.MinValue;
+            var contentDictionary = new Dictionary<string, IReadOnlyList<Tuple<Timestamp, string>>>();
+
+            foreach (var path in this.filePaths)
+            {
+                var list = new List<Tuple<Timestamp, string>>();
+                var content = System.IO.File.ReadAllLines(path);
+                
+
+                foreach (var line in content)
+                {
+                    var items = line.Split(new[] { ',' }, 2);
+
+                    if (items.Length < 2)
+                    {
+                        throw new InvalidOperationException("File line cannot be split to two sub-strings");
+                    }
+
+                    DateTime time;
+                    if (!DateTime.TryParse(items[0], out time))
+                    {
+                        throw new InvalidOperationException("Time cannot be pasred to DateTime format");
+                    }
+
+                    var timeStamp = Timestamp.FromNanoseconds(time.Ticks * 100);
+                    var words = items[1];
+
+                    if (timeStamp < startTime)
+                    {
+                        startTime = timeStamp;
+                        firstEvent = time;
+                    }
+
+                    if (timeStamp > endTime)
+                    {
+                        endTime = timeStamp;
+                    }
+
+                    list.Add(new Tuple<Timestamp, string>(timeStamp, words));
+                }
+
+                contentDictionary[path] = list.AsReadOnly();
+                this.dataSourceInfo = new DataSourceInfo(startTime.ToNanoseconds, endTime.ToNanoseconds, firstEvent.ToUniversalTime());
```

Comment:
why is this set for every file? can it be moved to after the for loop?

Reply:
This code should be updated, originally this processor was only supporting 1 file, should be updated to support many